code/game/objects/structures/crates_lockers/closets/secure/engineering.dm

```diff
@@ -15,6 +15,10 @@
 			new /obj/item/weapon/storage/backpack/industrial(src)
 		else
 			new /obj/item/weapon/storage/backpack/satchel_eng(src)
+		if(prob(50))
+			new /obj/item/weapon/storage/backpack/dufflebag/eng(src)
+		else
+			return

```

Comment:
This is probably why some lockers seem to have way less stuff than they should, since if the RNG roll fails, the game sees 'return' and stops filling the locker, so anything below it won't get filled.